dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -432,6 +308,19 @@ def _iter_dataset_record_ids(self, context: QueryContext):
                 )
             yield dim_type, dataset_record_ids
 
+    @staticmethod
+    def _list_dimension_columns(df: DataFrame) -> list[str]:
+        columns = DimensionType.get_allowed_dimension_column_names()
+        return [x for x in df.columns if x in columns]
+
+    def _list_dimensions(self, df: DataFrame) -> list[DimensionType]:
+        dims = [DimensionType(x) for x in DatasetSchemaHandlerBase._list_dimension_columns(df)]
+        if self._config.get_table_format_type() == TableFormatType.PIVOTED:
+            pivoted_type = self._config.get_pivoted_dimension_type()
+            assert pivoted_type is not None
+            dims.append(pivoted_type)
```

Comment:
Is this right? _list_dimensions() is used to retain relevant columns and make association table. If pivoted on metric, metric would not be a column name right?

Reply:
Do we have a test for this section?